README.md

```diff
@@ -102,10 +102,67 @@ If the `--info` flag is provided to gradle it will output a dependency tree whic
 
 ![Tree](docs/Tree.PNG)
 
-Report output
+## Reporting in a Jenkins Pipeline
 -------------
 
-TODO
+The gradle plugin supports writing out test reports in the correct XML format for the
+[Jenkins JUnit Reporting Plugin](https://wiki.jenkins.io/display/JENKINS/JUnit+Plugin).
+To switch on this reporting, set the path to the report in you project's build.gradle file using the **"junitReport"** element like so:
+
+```
+audit {
+        failOnError = false
+        ignore = [ 'ch.qos.logback:logback-core' ]        ]
+        junitReport = "./ossindex/junitReport.xml"
+    }
+```
+
+This would create the file in an /ossindex folder in the project root.
+
+To access this using the JUnit plugin in a Jenkins pipeline:
+
+```
+    stage('OSSIndex Scan') {
+        steps {
+            // Run the audit
+            sh "./gradlew --no-daemon --stacktrace audit"
+        }
+        post {
+            always {
+                // Tell junit plugin to use report
+                junit '**/ossindex/junitReport.xml'
+                // Fail stage if '<failure' found in report
+                sh "[[ ! \$(grep '<failure' ./ossindex/junitReport.xml) ]]"
+            }
+        }
+    }
+```
+
+NOTE: The junit plugin uses a slightly different syntax to reference the path.
+
+The example code creates a stage in the pipeline, best put between checkout and compile, to run the ossindex
+scan and then run the reporting plugin.
+
+The line:
+```
+    sh "[[ ! \$(grep '<failure' ./ossindex/junitReport.xml) ]]"
+```
+
+Ensures that the build fails if any failures are reported.
+
+Set
+```
+    failOnError = false
+```
+As failOnError is true by default and will cause the scan to exit on the first failure, instead of finding them all.

```

Comment:
Doesn't the `post` and `always` step act like a `finally` block, so this shouldn't be needed?

I use `try { runTests } finally { junit… }` in my builds to always record test results but still fail the build if a test failed, because the Exception bubbles up. 
I'm not familar with the `post always` semantic though

Reply:
Hi Markus,

It is necessary to explicitly set this false as it's true by default. This has the effect of failing the build immediately, which was a problem for me. I tested this on a rather large and ungainly Java project. When the gradle audit was run, it would split the audit out into multiple, 20+, sub projects. Each of which would then have its own sub audit. 

So any sub audits running after a failure would be skipped and we would not see the results.

The work to include this in the pipeline also saw me adding support for the Jenkins JUnit plugin which gives a nice report on all of the failures. 

So in summary, when in a pipeline, this needs to be false!

Brad

Reply:
Ah, I see :)